internal/sapsystem/sapcontrol/webservice.go

```diff
@@ -0,0 +1,150 @@
+}
+
+type STATECOLOR string
+type STATECOLOR_CODE int
+
+const (
+	STATECOLOR_GRAY        STATECOLOR      = "SAPControl-GRAY"
+	STATECOLOR_GREEN       STATECOLOR      = "SAPControl-GREEN"
+	STATECOLOR_YELLOW      STATECOLOR      = "SAPControl-YELLOW"
+	STATECOLOR_RED         STATECOLOR      = "SAPControl-RED"
+	STATECOLOR_CODE_GRAY   STATECOLOR_CODE = 1
+	STATECOLOR_CODE_GREEN  STATECOLOR_CODE = 2
+	STATECOLOR_CODE_YELLOW STATECOLOR_CODE = 3
+	STATECOLOR_CODE_RED    STATECOLOR_CODE = 4
+)
+
+type GetInstanceProperties struct {
+	XMLName xml.Name `xml:"urn:SAPControl GetInstanceProperties"`
+}
+
+type GetProcessList struct {
+	XMLName xml.Name `xml:"urn:SAPControl GetProcessList"`
+}
+
+type GetProcessListResponse struct {
+	XMLName   xml.Name     `xml:"urn:SAPControl GetProcessListResponse"`
+	Processes []*OSProcess `xml:"process>item,omitempty" json:"process>item,omitempty"`
+}
+type GetInstancePropertiesResponse struct {
+	XMLName    xml.Name            `xml:"urn:SAPControl GetInstancePropertiesResponse"`
+	Properties []*InstanceProperty `xml:"properties>item,omitempty" json:"properties>item,omitempty"`
+}
+
+type GetSystemInstanceList struct {
+	XMLName xml.Name `xml:"urn:SAPControl GetSystemInstanceList"`
+	Timeout int32    `xml:"timeout,omitempty" json:"timeout,omitempty"`
+}
+
+type GetSystemInstanceListResponse struct {
+	XMLName   xml.Name       `xml:"urn:SAPControl GetSystemInstanceListResponse"`
+	Instances []*SAPInstance `xml:"instance>item,omitempty" json:"instance>item,omitempty"`
+}
+
+type OSProcess struct {
+	Name        string     `xml:"name,omitempty" json:"name,omitempty" mapstructure:"name,omitempty"`
+	Description string     `xml:"description,omitempty" json:"description,omitempty" mapstructure:"description,omitempty"`
+	Dispstatus  STATECOLOR `xml:"dispstatus,omitempty" json:"dispstatus,omitempty" mapstructure:"dispstatus,omitempty"`
+	Textstatus  string     `xml:"textstatus,omitempty" json:"textstatus,omitempty" mapstructure:"textstatus,omitempty"`
+	Starttime   string     `xml:"starttime,omitempty" json:"starttime,omitempty" mapstructure:"starttime,omitempty"`
+	Elapsedtime string     `xml:"elapsedtime,omitempty" json:"elapsedtime,omitempty" mapstructure:"elapsedtime,omitempty"`
+	Pid         int32      `xml:"pid,omitempty" json:"pid,omitempty" mapstructure:"pid,omitempty"`
+}
+
+type InstanceProperty struct {
+	Property     string `xml:"property,omitempty" json:"property,omitempty" mapstructure:"property,omitempty"`
+	Propertytype string `xml:"propertytype,omitempty" json:"propertytype,omitempty" mapstructure:"propertytype,omitempty"`
+	Value        string `xml:"value,omitempty" json:"value,omitempty" mapstructure:"value,omitempty"`
+}
+
+type SAPInstance struct {
+	Hostname      string     `xml:"hostname,omitempty" json:"hostname,omitempty" mapstructure:"hostname,omitempty"`
+	InstanceNr    int32      `xml:"instanceNr,omitempty" json:"instanceNr,omitempty" mapstructure:"instanceNr,omitempty"`
```

Comment:
In `mapstructure` annotations, to keep consistency, I would rename the values to use only lowercase.
Example: `instanceNr` -> `Instancenr` and `httpPort` -> `httpport`